### DISCUSSION TITLE
[5.x] Option for anchor in all link field types

### DISCUSSION THREAD

**Discussion by MangoMarcus** [craftcms/cms#15813] [Ideas] [answered]
The new link field type in Craft 5.3 includes an option to "allow anchors" but this seems to only apply to "URL" link types.

It would be good to also specify an anchor (ie. url fragment) for other link types, eg. entries.

The abandoned "typed link" plugin included an option for this labelled "Allow users to enter custom queries that will be appended to the url", which just rendered a free text field

### Typed link field settings
<img width="493" alt="Screenshot 2024-09-30 at 19 25 09" src="https://github.com/user-attachments/assets/d3b4f397-d6c9-45c5-808e-44670bea6ae0">

### Typed link fragment field
<img width="808" alt="Screenshot 2024-09-30 at 19 26 44" src="https://github.com/user-attachments/assets/6622426a-18f0-4fb3-ae1a-99233771c942">

**Comment by MangoMarcus** [comment]
Just to add, Hyper also supports this with the "URL suffix" field

<img width="847" alt="Screenshot 2024-11-14 at 14 27 36" src="https://github.com/user-attachments/assets/33b12d0b-637a-4c96-a919-a31f1372a57f">

**Comment by brandonkelly** [comment, marked as answer]
Added for Craft 5.6 (#16139) 🎉

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀

**Comment by MangoMarcus** [comment]
Awesome! Ty 🙏
